yarn-project/acir-simulator/src/avm/serialization/instruction_serialization.ts

```diff
@@ -0,0 +1,132 @@
+import { BufferCursor } from './buffer_cursor.js';
+
+/**
+ * All AVM opcodes. (Keep in sync with cpp counterpart code AvmMini_opcode.hpp).
+ * Source: https://yp-aztec.netlify.app/docs/public-vm/instruction-set
+ */
+export enum Opcode {
+  ADD = 0x00,
+  SUB = 0x01,
+  MUL = 0x02,
+  DIV = 0x03,
+  EQ = 0x04,
+  LT = 0x05,
+  LTE = 0x06,
+  AND = 0x07,
+  OR = 0x08,
+  XOR = 0x09,
+  NOT = 0x0a,
+  SHL = 0x0b,
```

Comment:
Do we need to specify these all at this stage? They're certainly going to change over the coming months and we might be more likely to make a mistake if we need to update these versus just inserting/removing one and letting the following ones automatically update.